src/detect.c

```diff
@@ -808,7 +808,7 @@ static inline void DetectRulePacketRules(
 #endif
         DetectRunPostMatch(tv, det_ctx, p, s);
 
-        AlertQueueAppend(det_ctx, s, p, 0, alert_flags);
+        AlertQueueAppend(det_ctx, s, p, PACKET_ALERT_NOTX, alert_flags);
```

Comment:
Should we try to get a tx id in some cases ? (cf SV PR)
And set `PACKET_ALERT_FLAG_TX` ?

Reply:
I think we may have to if the match is in the stream. I suppose we should then log the most recent tx, not just tx id 0

Reply:
Done in next version

But this is not only "stream" match, as we should also do it for UDP...